ykrt/src/compile/jitc_yk/codegen/x64/mod.rs

```diff
@@ -1328,14 +1319,41 @@ impl<'a> Assemble<'a> {
         Ok(())
     }
 
-    fn cg_icmp(&mut self, iidx: InstIdx, inst: &jit_ir::ICmpInst) {
-        let (lhs, pred, rhs) = (inst.lhs(self.m), inst.predicate(), inst.rhs(self.m));
-        let bit_size = self.m.type_(lhs.tyidx(self.m)).bit_size().unwrap();
-        let [lhs_reg, rhs_reg] = self.ra.assign_gp_regs(
-            &mut self.asm,
-            iidx,
-            [RegConstraint::InputOutput(lhs), RegConstraint::Input(rhs)],
-        );
+    /// If an `Operand` refers to a constant integer that can be represented as an `i32`, return
+    /// it, otherwise return `None`.
+    fn op_to_i32(&self, op: &Operand) -> Option<i32> {
+        if let Operand::Const(cidx) = op {
+            if let Const::Int(tyidx, v) = self.m.const_(*cidx) {
+                let Ty::Integer(bit_size) = self.m.type_(*tyidx) else {
+                    panic!()
+                };
+                if *bit_size <= 32 {
```

Comment:
Nice. This enables more optimisations than before, since we don't check for the bit size being exactly 32-bits like we did before. Anything that fits in 32-bits should be fine!

It was on my list to look at this, so I'm happy you've done it for me :)